rasa/projects/openWHO/actions/actions.py

```diff
@@ -0,0 +1,82 @@
+from typing import Text, Dict, Any, List
+from rasa_sdk import Action, Tracker
+from rasa_sdk.events import SlotSet, SessionStarted
+from sanic.request import Request
+from rasa_sdk.executor import CollectingDispatcher
+
+import requests
+import json
+
+class CourseSet(Action):
+	def name(self):
+		return "action_course_set"
+
+	def run(self, dispatcher, tracker, domain):
+		currentCourse = tracker.get_slot('current_course')
+		if currentCourse:
+			return [SlotSet('course-set', True)]
+		else:
+			return [SlotSet('course-set', False)]
+
+class SetCurrentCourse(Action):
+	def name(self):
+		return "action_set_current_course"
+
+	def run(self, dispatcher, tracker, domain):
+		currentCourse = tracker.latest_message['text']
+		return [SlotSet('current_course', currentCourse)]
+
+class ActionGetCourses(Action):
+	def name(self) -> Text:
+		return "action_get_courses"
+
+	def run(self, dispatcher: CollectingDispatcher, tracker: Tracker, domain: Dict[Text, Any]) -> List[Dict[Text, Any]]:
+		current_state = tracker.current_state()
+		token = current_state['sender_id']
+		r = requests.get('http://localhost:3000/bridges/chatbot/my_courses', headers={"content-type": "application/json",
```

Comment:
shouldn't this come from a config?

Reply:
the token?

Reply:
As the token changes from session to session slots are the better way to proceed here. Slots will be removed after the session has ended or 60 minutes